pkg/ccl/backupccl/backup_test.go

```diff
@@ -1443,6 +1443,252 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
+				tableName := "t1"
+				createTable(sqlDB, tableName)
+				sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_publishing_descriptors'`)
+
+				// t0: Backup a table that we'll then restore with pause
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO '%s'`, localFoo))
+				sqlDB.Exec(t, fmt.Sprintf(`DROP TABLE %s`, tableName))
+				var restoreJobID int
+				sqlDB.QueryRow(t, fmt.Sprintf(`RESTORE TABLE %s FROM LATEST IN '%s' with detached`, tableName, localFoo)).Scan(&restoreJobID)
+				jobutils.WaitForJobToPause(t, sqlDB, jobspb.JobID(restoreJobID))
+
+				// t1: run an inc backup while restore is paused to ensure the introduced span contains data in the subseq
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO LATEST IN '%s'`, localFoo))
+
+				// t2: unpause the restore, allowing the next incremental backup to capture its reintroduced spans.
+				sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = ''`)
+				sqlDB.Exec(t, `RESUME JOB $1`, restoreJobID)
+				jobutils.WaitForJobToSucceed(t, sqlDB, jobspb.JobID(restoreJobID))
+			},
+			checkpointsIntroducedSpans: true,
+		},
+	} {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+
+			ctx := context.Background()
+
+			// We turn on the following testing knobs during the backup whose
+			// checkpoints we want to track.k
+			exportedSpanCount := make(map[string]int)
+			var logExportSpanCount bool
+			var holdUpCheckpointing bool
+			waitForProgress := make(chan struct{})
+			waitForPause := make(chan struct{})
+
+			params := base.TestClusterArgs{}
+			var mu syncutil.Mutex
+			knobs := base.TestingKnobs{
+				SQLExecutor: &sql.ExecutorTestingKnobs{
+					AfterBackupCheckpoint: func() {
+						mu.Lock()
+						if holdUpCheckpointing && len(exportedSpanCount) > 0 {
+							close(waitForProgress)
+							<-waitForPause
+							holdUpCheckpointing = false
+						}
+						mu.Unlock()
+					},
+				},
+				DistSQL: &execinfra.TestingKnobs{
+					BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
+						RunAfterExportingSpanEntry: func(ctx context.Context, response *kvpb.ExportResponse) {
+							mu.Lock()
+							if logExportSpanCount {
+								if response.Files != nil && len(response.Files) > 0 {
+									exportedSpan := response.Files[0].Span.String()
+									exportedSpanCount[exportedSpan] += 1
+								}
+							}
+							mu.Unlock()
+						},
+					},
+				},
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			}
+			params.ServerArgs = base.TestServerArgs{Knobs: knobs}
+
+			tc, sqlDB, dir, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, 1,
+				InitManualReplication, params)
+			defer cleanupFn()
+
+			sqlDB.Exec(t, `CREATE DATABASE d`)
+			sqlDB.Exec(t, "USE d")
+
+			testCase.preBackupWorkload(ctx, t, sqlDB)
+
+			logExportSpanCount = true
+			holdUpCheckpointing = true
+			sqlDB.Exec(t, "SET CLUSTER SETTING bulkio.backup.checkpoint_interval = '10ms'")
```

Comment:
🤔 Over in the IMPORT processor, I think we have a testing hook where we checkpoint on every update. I wonder if that would be useful in some tests. But, even so, I wouldn't do it here because it is nice to be able to backport this with very few non-test code changes.

Reply:
Agreed. I don't want to add that testing hook.